### PR TITLE
Add documentation about reloading search analyzers

### PR DESCRIPTION
**Pull request by cbuescher**
This PR adds documentation to the feature being worked on in #42888

### CODE REVIEW

File: docs/reference/indices/reload-analyzers.asciidoc [hunk 1]
```diff
@@ -0,0 +1,56 @@
+[[indices-reload-analyzers]]
```
Comment (cbuescher) [inline docs/reference/indices/reload-analyzers.asciidoc]: jimczi on #42865: This should be marked as experimental.

File: docs/reference/analysis/tokenfilters/synonym-tokenfilter.asciidoc [hunk 2]
```diff
@@ -181,3 +185,40 @@ error.
...
+        }
+    }
+}
+--------------------------------------------------
+// CONSOLE
+
+Using the <<analyzer reload API,indices-reload-analyzers>>, you can trigger reloading of the
+synonym definition. The contents of the configured synonyms file will be reloaded and the
+synonyms definition the filter uses will be updated. Note that if you trying to use the above
+analyzer as an index analyzer will result in an error.
```
Comment (cbuescher) [inline docs/reference/analysis/tokenfilters/synonym-tokenfilter.asciidoc]: jimczi on #42865: I would extract the last sentence in a NOTE in order to make sure that users are aware of the limitation.

File: docs/reference/indices/reload-analyzers.asciidoc [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    }
+}
+--------------------------------------------------
+// CONSOLE
+
+<1> Mark the synonym filter as updateable.
+<2> Synonym filter is usable as a search_analyzer.
+
+Calling the `_reload_search_analyzers` endpoint will now trigger reloading the
+synonyms from the configured "synonym.txt" file.
```
Comment (cbuescher) [inline docs/reference/indices/reload-analyzers.asciidoc]: jimczi on #42865: 

Maybe add a NOTE regarding the fact that the reload is per node so you need to update your synonym.txt file on every node before calling reload.
@cbuescher

**Comment by cbuescher** [comment]
@jimczi sorry again for messing up the previous feature branch, I transfered your comments from #42865 and hopefully addressed them with my following commit.

**Comment by cbuescher** [comment]
Just to note, the docs checks cannot pass as long as #42888 which contains the implemention for this is merged into the feature branch. I was thinking about cherry-picking those changes here but then that would make keeping this branch up-to date more difficult, so I'm disregarding docs checks for the time being.

**Comment by cbuescher** [comment]
@elasticmachien update branch

**Comment by cbuescher** [comment]
@elasticmachine update branch

**Comment by cbuescher** [comment]
@jimczi tests pass now that #42888 is merged to the feature branch. Would you mind taking another look? I'd like to merge this as well and open a PR for the whole feature branch to get final reviews there.

File: docs/reference/analysis/tokenfilters/synonym-tokenfilter.asciidoc [hunk 4]
```diff
@@ -181,3 +186,41 @@ error.
...
+    }
+}
+--------------------------------------------------
+// CONSOLE
+
+Using the <<indices-reload-analyzers,analyzer reload API>>, you can trigger reloading of the
+synonym definition. The contents of the configured synonyms file will be reloaded and the
+synonyms definition the filter uses will be updated. 
+
+Note: Trying to use the above analyzer as an index analyzer will result in an error.
```
Comment (jimczi) [inline docs/reference/analysis/tokenfilters/synonym-tokenfilter.asciidoc]: nit: s/Note:/NOTE:/ ?

File: docs/reference/indices/reload-analyzers.asciidoc [hunk 5]
```diff
@@ -0,0 +1,65 @@
+[[indices-reload-analyzers]]
+== Reload Search Analyzers
+
+experimental[]
+
+Reloads search analyzers and its resources.
+
+The `_reload_search_analyzers` API can be run on one more indices and will
```
Comment (jimczi) [inline docs/reference/indices/reload-analyzers.asciidoc]: nit:/can be run on one more/can be run on one or more/

File: docs/reference/indices/reload-analyzers.asciidoc [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+        }
+    }
+}
+--------------------------------------------------
+// CONSOLE
+
+<1> Mark the synonym filter as updateable.
+<2> Synonym analyzer is usable as a search_analyzer.
+
+Calling the `_reload_search_analyzers` endpoint will now trigger reloading the
```
Comment (jimczi) [inline docs/reference/indices/reload-analyzers.asciidoc]: nit:s/reloading the/reloading of the/ ?

**Review by jimczi — APPROVED** [review]
I left some nits, LGTM otherwise

**Pull request merged**